nibabel/freesurfer/io.py

```diff
@@ -44,21 +45,64 @@ def _fread3_many(fobj, n):
     return (b1 << 16) + (b2 << 8) + b3
 
 
-def read_geometry(filepath):
+def _read_volume_info(extra):
+    volume_info = OrderedDict()
+
+    if extra is None:
+        return volume_info
+
+    if extra[:4] != b'\x00\x00\x00\x14':
+        warnings.warn("Unknown extension code.")
+    else:
+        try:
+            for line in extra[4:].split(b'\n'):
+                if len(line) == 0:
+                    continue
+                key, val = map(bytes.strip, line.split(b'=', 1))
+                key = key.decode('utf-8')
+                if key in ('voxelsize', 'xras', 'yras', 'zras', 'cras'):
+                    val = np.fromstring(val, sep=' ')
+                    val = val.astype(np.float)
+                elif key == 'volume':
+                    val = np.fromstring(val, sep=' ', dtype=np.uint)
+                    val = val.astype(np.int)
+                volume_info[key] = val
+        except ValueError:
+            raise ValueError("Error parsing volume info")
+
+    if len(volume_info) == 0:
+        warnings.warn("Volume geometry info is either "
+                      "not contained or not valid.")
```

Comment:
I don't think we need to warn if there is no volume info, that's clear from the dict being returned as empty right? Plus this ends up being redundant with the `warnings.warn()` call above, so at the very least this should go in the `else` clause above if you want to keep it (and the "unknown extension code" warning should probably have some of the same text, but come out in a single warning instead of 2 as it would be here).